src/representation_theory/basic.lean

```diff
@@ -7,105 +7,196 @@ import algebra.module.basic
 import algebra.module.linear_map
 import algebra.monoid_algebra.basic
 import linear_algebra.trace
+import linear_algebra.dual
+import linear_algebra.free_module.basic
 
 /-!
 # Monoid representations
 
-This file introduces monoid representations and their characters and proves basic lemmas about them,
-including equivalences between different definitions of representations.
+This file introduces monoid representations and their characters and defines a few ways to construct
+representations.
 
 ## Main definitions
 
-  * `representation.as_module`
-  * `representation.as_group_hom`
-  * `representation.character`
+  * representation.representation
+  * representation.character
+  * representation.tprod
+  * representation.lin_hom
+  * represensation.dual
 
 ## Implementation notes
 
-A representation of a monoid `G` over a commutative semiring `k` is implemented as a `k`-module `V`
-together with a `distrib_mul_action G V` instance and a `smul_comm_class G k V` instance.
-
-Alternatively, one can use a monoid homomorphism `G →* (V →ₗ[k] V)`. The definitions `as_monoid_hom`
-and `rep_space` allow to go back and forth between these two definitions.
+Representations of a monoid `G` on a `k`-module `V` are implemented as
+homomorphisms `G →* (V →ₗ[k] V)`.
 -/
 
-open monoid_algebra
+open monoid_algebra (lift) (of)
+open linear_map
 
 namespace representation
 
 section
-variables (k G V : Type*) [comm_semiring k] [monoid G] [add_comm_monoid V]
-variables [module k V] [distrib_mul_action G V] [smul_comm_class G k V]
+variables (k G V : Type*) [comm_semiring k] [monoid G] [add_comm_monoid V] [module k V]
+
+/--
+A representation of `G` on the `k`-module `V` is an homomorphism `G →* (V →ₗ[k] V)`.
+-/
+@[nolint dup_namespace]
```

Comment:
I suspect this lint error is something you do want to fix - right now your type is called `representation.representation`, which will be annoying, and prevents dot notation.

Reply:
Should I just put the abbreviation out of the namespace?

Reply:
Yes; either by moving it, or by adding `_root_`.